File: frontend/src/api/adminStorage.ts
import { apiClient } from '@/api/client'

const PREFIX = '/api/v1/admin/storage'

export interface StorageConfigAdmin {
  storage_type: string
  local_storage_path: string | null
  local_serve_prefix: string | null
  s3_bucket: string | null
  s3_region: string | null
  s3_endpoint_url: string | null
  s3_access_key: string | null
  s3_public_base_url: string | null
  image_upload_max_bytes: number
  public_access: boolean
  is_active: boolean
  secret_configured: boolean
  updated_at: string | null
}

export interface UpdateStorageConfigPayload {
  storage_type: string
  local_storage_path?: string | null
  local_serve_prefix?: string | null
  s3_bucket?: string | null
  s3_region?: string | null
  s3_endpoint_url?: string | null
  s3_access_key?: string | null
  s3_secret_key?: string | null
  s3_public_base_url?: string | null
  image_upload_max_bytes?: number
  public_access?: boolean
  is_active?: boolean
}

export interface StorageTestResult {
  status: string
  message: string
}

export const adminStorageApi = {
  async get(): Promise<StorageConfigAdmin> {
    return apiClient.get<StorageConfigAdmin>(PREFIX)
  },

  async update(payload: UpdateStorageConfigPayload): Promise<StorageConfigAdmin> {
    return apiClient.put<StorageConfigAdmin>(PREFIX, payload)
  },

  async testConnection(): Promise<StorageTestResult> {
    return apiClient.post<StorageTestResult>(`${PREFIX}/test`, {})
  },
}

File: backend/domains/agent/presentation/schemas/admin_storage.py
"""Admin Storage API schemas。"""

from __future__ import annotations

from pydantic import BaseModel, Field


class StorageConfigAdminResponse(BaseModel):
    storage_type: str
    local_storage_path: str | None = None
    local_serve_prefix: str | None = None
    s3_bucket: str | None = None
    s3_region: str | None = None
    s3_endpoint_url: str | None = None
    s3_access_key: str | None = None
    s3_public_base_url: str | None = None
    image_upload_max_bytes: int
    public_access: bool
    is_active: bool
    secret_configured: bool
    updated_at: str | None = None


class UpdateStorageConfigBody(BaseModel):
    storage_type: str = Field(..., description="local | s3")
    local_storage_path: str | None = None
    local_serve_prefix: str | None = None
    s3_bucket: str | None = None
    s3_region: str | None = None
    s3_endpoint_url: str | None = None
    s3_access_key: str | None = None
    s3_secret_key: str | None = Field(
        None,
        description="留空表示不修改已有 secret",
    )
    s3_public_base_url: str | None = None
    image_upload_max_bytes: int = Field(10_485_760, ge=1)
    public_access: bool = True
    is_active: bool = True


class StorageTestResponse(BaseModel):
    status: str
    message: str


__all__ = [
    "StorageConfigAdminResponse",
    "StorageTestResponse",
    "UpdateStorageConfigBody",
]
